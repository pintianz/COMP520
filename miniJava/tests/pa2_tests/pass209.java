//PA2 pass qualified reference assignment
class A {
    void p(){
        x.y[3] = 3;
    }
}
