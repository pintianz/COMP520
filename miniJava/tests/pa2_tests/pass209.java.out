======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "p" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . AssignStmt
  .   .   .   IndexedRef
  .   .   .     LiteralExpr
  .   .   .       "3" IntLiteral
  .   .   .     QualifiedRef
  .   .   .       "y" Identifier
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .   LiteralExpr
  .   .   .     "3" IntLiteral
=============================================
